tests/unit/s2n_kem_test.c

```diff
@@ -0,0 +1,141 @@
+#define TEST_CIPHERTEXT_LENGTH  5
+const char TEST_CIPHERTEXT[] = {5, 5, 5, 5, 5};
+
+
+int s2n_test_generate_keypair(unsigned char *public_key, unsigned char *private_key)
+{
+    memset(public_key, TEST_PUBLIC_KEY_LENGTH, TEST_PUBLIC_KEY_LENGTH);
+    memset(private_key, TEST_PRIVATE_KEY_LENGTH, TEST_PRIVATE_KEY_LENGTH);
+    return 0;
+}
+int s2n_test_encrypt(unsigned char *ciphertext, unsigned char *shared_secret, const unsigned char *public_key)
+{
+    GUARD(memcmp(public_key, TEST_PUBLIC_KEY, TEST_PUBLIC_KEY_LENGTH));
+    memset(ciphertext, TEST_CIPHERTEXT_LENGTH, TEST_CIPHERTEXT_LENGTH);
+    memset(shared_secret, TEST_SHARED_SECRET_LENGTH, TEST_SHARED_SECRET_LENGTH);
+    return 0;
+}
+int s2n_test_decrypt(unsigned char *shared_secret, const unsigned char *ciphertext, const unsigned char *private_key)
+{
+    GUARD(memcmp(ciphertext, TEST_CIPHERTEXT, TEST_CIPHERTEXT_LENGTH));
+    GUARD(memcmp(private_key, TEST_PRIVATE_KEY, TEST_PRIVATE_KEY_LENGTH));
+    memset(shared_secret, TEST_SHARED_SECRET_LENGTH, TEST_SHARED_SECRET_LENGTH);
+    return 0;
+}
+
+const struct s2n_kem s2n_test_kem = {
+        .public_key_length = TEST_PUBLIC_KEY_LENGTH,
+        .private_key_length = TEST_PRIVATE_KEY_LENGTH,
+        .shared_secret_key_length = TEST_SHARED_SECRET_LENGTH,
+        .ciphertext_length = TEST_CIPHERTEXT_LENGTH,
+        .generate_keypair = &s2n_test_generate_keypair,
+        .encapsulate = &s2n_test_encrypt,
+        .decapsulate = &s2n_test_decrypt,
+};
+
+int main(int argc, char **argv)
+{
+    BEGIN_TEST();
+    {
+        // regression test for network parsing data of expected sizes
+        EXPECT_EQUAL(sizeof(kem_extension_size), 1);
+        EXPECT_EQUAL(sizeof(kem_public_key_size), 2);
+        EXPECT_EQUAL(sizeof(kem_private_key_size), 2);
+        EXPECT_EQUAL(sizeof(kem_shared_secret_size), 2);
+        EXPECT_EQUAL(sizeof(kem_ciphertext_key_size), 2);
+    }
+    {
+        struct s2n_kem_params server_params = {0};
+        server_params.negotiated_kem = &s2n_test_kem;
+        EXPECT_SUCCESS(s2n_kem_generate_keypair(&server_params));
+        EXPECT_EQUAL(TEST_PUBLIC_KEY_LENGTH, server_params.public_key.size);
+        EXPECT_EQUAL(TEST_PRIVATE_KEY_LENGTH, server_params.private_key.size);
+        EXPECT_BYTEARRAY_EQUAL(TEST_PUBLIC_KEY, server_params.public_key.data, TEST_PUBLIC_KEY_LENGTH);
+        EXPECT_BYTEARRAY_EQUAL(TEST_PRIVATE_KEY, server_params.private_key.data, TEST_PRIVATE_KEY_LENGTH);
+
+
+        struct s2n_kem_params client_params = {0};
+        client_params.negotiated_kem = &s2n_test_kem;
+        // This would be handled by client/server key exchange methods which isn't being tested
+        GUARD(s2n_alloc(&client_params.public_key, TEST_PUBLIC_KEY_LENGTH));
+        memset(client_params.public_key.data, TEST_PUBLIC_KEY_LENGTH, TEST_PUBLIC_KEY_LENGTH);
+
+        struct s2n_blob client_shared_secret = {0};
+        struct s2n_blob ciphertext = {0};
+        EXPECT_SUCCESS(
```

Comment:
nitpick: I would make this one line since you already have lines that are longer than this would be

Reply:
Fixed, that might have been a typo or my IDE.